heron/scheduler-core/src/java/com/twitter/heron/scheduler/dryrun/JsonFormatterUtils.java

```diff
@@ -0,0 +1,87 @@
+//  Copyright 2018 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.scheduler.dryrun;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.packing.Resource;
+
+public class JsonFormatterUtils {
+
+  private final ObjectMapper mapper;
```

Comment:
I am wondering if ObjectWriter is available to us, as well as the how to reuse this object.

Reply:
Thx for the reply. I wasn't too worry about the withDefaultPrettyPrinter() function.

It just seems like ObjectWriter and reusing are preferred based on some ObjectMapper document, so I brought the question up. ObjectMapper is kinda magical to me. Hopefully some of us have more experience with it could provide more feedbacks. :)

Reply:
FWIW here's a comment on StackOverflow regarding re-usability.

>It's fine to use a single instance per application provided you don't call any configuration methods after it's been made visible i.e. you should do all your initialization inside a static block.
https://stackoverflow.com/questions/18611565/how-do-i-correctly-reuse-jackson-objectmapper

Given that the documentation mentions that it's thread-safe and I am not storing any of the intermediate steps in class variables I think this should be safe from both a re-usability standpoint and thread-safety standpoint as written but I can defer to someone who has more jackson experience to weigh in as well.

Reply:
Sounds good to me. thanks~

Reply:
ObjectMapper is usually used for a single dump call, while ObjectWriter is used for cases with intermediates steps. Re-using single ObjectMapper should be fine in this case based on my understanding. We have some internal and open source codes doing this kind of single object reusing.

Reply:
kk. Thx for the input.

I am going to approve it and you can decide when to submit.